lib/checkother.cpp

```diff
@@ -1391,6 +1391,16 @@ void CheckOther::checkConstVariable()
         // Skip if address is taken
         if (Token::findmatch(var->nameToken(), "& %varid%", scope->bodyEnd, var->declarationId()))
             continue;
+        // Skip if another non-const variable is initialized with this variable
+        {
+            auto match = Token::findmatch(var->nameToken(), "%type% & %name% = %varid%", scope->bodyEnd, var->declarationId());
```

Comment:
maybe it would be better to loop through all tokens. imagine that there is first a const reference initialization and then a non-const initialization. This code will only check the first initialization.

Reply:
I've extended my tests to include this case and indeed it currently fails.